xnmt/input_reader.py

```diff
@@ -122,37 +124,46 @@ def count_words(self, trg_words):
   def vocab_size(self):
     return len(self.vocab)
 
-class BaseTextReader(InputReader):
+class SubwordSampleTextReader(BaseTextReader, Serializable):
   """
-  A base class for text-based :class:`xnmt.input_reader.InputReader` subclasses that implements some helper methods.
+  Handles the sampling for subword regularization.
+  https://arxiv.org/pdf/1804.10959.pdf 
   """
-  def count_sents(self, filename):
-    f = open(filename, encoding='utf-8')
-    try:
-      return sum(1 for _ in f)
-    finally:
-      f.close()
+  yaml_tag = '!SubwordSampleTextReader'
+
+  @serializable_init
+  def __init__(self, model_file, vocab=None, include_vocab_reference=False):
+    self.subword_model = spm.SentencePieceProcessor()
+    self.subword_model.Load(model_file)
+    self.vocab = vocab
+    self.include_vocab_reference = include_vocab_reference
+    if vocab is not None:
+      self.vocab.freeze()
+      self.vocab.set_unk(Vocab.UNK_STR)
+
+  def read_sent(self, sentence, filter_ids=None):
+    vocab_reference = self.vocab if self.include_vocab_reference else None
+    words = self.subword_model.SampleEncode(sentence.strip(), -1, 0.1)
```

Comment:
`-1` and `0.1` are parameters, right? Could you make these configurable by adding them to the constructor?